src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -86,23 +89,41 @@ func (r *StatefulSetResource) runUpdate(
 		return nil
 	}
 
-	if err = r.updateRestartingStatus(ctx, true); err != nil {
-		return fmt.Errorf("unable to turn on restarting status in cluster custom resource: %w", err)
+	if !r.getRestartingStatus() {
```

Comment:
The logic inside `updateRestartingStatus` already checks if cluster status needs to be updated. Why `runUpdate` would have this check? I would propose to move the restarting status in `MarkPodsForUpdate` function. There would be one less indentation. 

REF:
https://github.com/redpanda-data/redpanda/blob/ba8ddbbc43d3486f6f2b069191eace789f449e01/src/go/k8s/pkg/resources/statefulset_update.go#L526

Reply:
I don't want to move that there at this time. It would make updateRestartingStatus would behave differently  depending on whether the parameter was true or false.